tikv-client/src/main/java/com/pingcap/tikv/expression/visitor/PartitionLocator.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2022 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pingcap.tikv.expression.visitor;
+
+import static com.pingcap.tikv.expression.FuncCallExpr.Type.YEAR;
+
+import com.pingcap.tikv.expression.ColumnRef;
+import com.pingcap.tikv.expression.ComparisonBinaryExpression;
+import com.pingcap.tikv.expression.ComparisonBinaryExpression.Operator;
+import com.pingcap.tikv.expression.Constant;
+import com.pingcap.tikv.expression.Expression;
+import com.pingcap.tikv.expression.FuncCallExpr;
+import com.pingcap.tikv.expression.LogicalBinaryExpression;
+import com.pingcap.tikv.expression.LogicalBinaryExpression.Type;
+import com.pingcap.tikv.meta.TiTableInfo;
+import com.pingcap.tikv.partition.PartitionedTable.PartitionLocatorContext;
+import com.pingcap.tikv.row.Row;
+import com.pingcap.tikv.types.DateType;
+import com.pingcap.tikv.types.IntegerType;
+import java.sql.Date;
+import java.sql.Timestamp;
+
+public class PartitionLocator extends DefaultVisitor<Boolean, PartitionLocatorContext> {
```

Comment:
This locator is only for range partition and rangecolumns partition, maybe we can make the name more explicit for this?